internal/controller/storageclaim_controller.go

```diff
@@ -109,12 +114,45 @@ func (r *StorageClaimReconciler) SetupWithManager(mgr ctrl.Manager) error {
 	}); err != nil {
 		return fmt.Errorf("unable to set up FieldIndexer for VSC csi driver name: %v", err)
 	}
+	enqueueStorageClaim := handler.EnqueueRequestsFromMapFunc(
+		func(context context.Context, obj client.Object) []reconcile.Request {
+			sclaims := &v1alpha1.StorageClaimList{}
+			err := r.Client.List(context, sclaims, &client.ListOptions{Namespace: obj.GetNamespace()})
```

Comment:
storageClaim is a clusterscoped resource, also could you use the r.list function that we have?